zjit/src/hir.rs

```diff
@@ -3112,6 +3107,18 @@ mod opt_tests {
         "#]]);
     }
 
+    #[test]
+    fn test_rest_param_get_bb_param() {
+        eval("
+            def rest(*array) = array
+        ");
+        assert_optimized_method_hir("rest", expect![[r#"
+            fn rest:
+            bb0(v0:BasicObject):
```

Comment:
I wonder if we can (easily) add type information here. Is it always an exact Array?

Reply:
👍🏼 done in [90cfe46](https://github.com/ruby/ruby/pull/13276/commits/90cfe46e8cd3c96c93baf75d91505b0a27413335) 

(weird ripple: keyword rest is not always a HashExact since it could be nil when anonymous)